### PR TITLE
Migrate to vite and vitest

### PR DESCRIPTION
**Pull request by vkozio**
Part of #836

Build: **vite** build implemented for 
- examples/advanced
- modules/*

Testing: jest migrated to **vitest**
Coverage by @vitest/coverage-c8

React 17, added related peerdeps

Feedback is welcome

### CODE REVIEW

File: examples/advanced/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  defineConfig({
+    resolve: { alias },
+    root: 'src',
+    server: { open: true },
+  })
+);
+// NOTE: To use this example standalone (e.g. outside of deck.gl repo)
+// delete the local development overrides at the bottom of this file
+
+/*
```
Comment (ibgreen) [inline examples/advanced/vite.config.ts:25]: Delete commented code, or add comment explaining why we are keeping it.

File: examples/vite.config.local.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      seer: resolve(LIB_DIR, './node_modules/seer'),
+      react: resolve(LIB_DIR, './node_modules/react'),
+    },
+  };
+}
+// Optional: Enables reading mapbox token from environment variable
+// plugins: [new webpack.EnvironmentPlugin(['MapboxAccessToken'])],
+
+/*
+function addLocalDevSettings(config, exampleDir) {
```
Comment (ibgreen) [inline examples/vite.config.local.ts:61]: Ditto

File: jest.config.js [hunk 3]
```diff
@@ -1,9 +0,0 @@
-require('core-js/stable');
-
-module.exports = {
```
Comment (ibgreen) [inline jest.config.js:3]: Why is this removed?

File: modules/edit-modes/test/lib/__snapshots__/measure-angle-mode.test.ts.snap [hunk 4]
```diff
@@ -1,9 +1,9 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
+// Vitest Snapshot v1
 
-exports[`three clicks + pointer move tooltip contains angle 1`] = `
-Array [
```
Comment (ibgreen) [inline modules/edit-modes/test/lib/__snapshots__/measure-angle-mode.test.ts.snap:4]: Why is this changing? Some jest version upgrade? Can this be handled in a separate PR?

File: vitest.config.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+// require('core-js/stable');
+
+// module.exports = {
```
Comment (ibgreen) [inline vitest.config.ts:3]: Cut commented code.

**Review by ibgreen — APPROVED** [review]

File: jest.config.js (hunk 3 above)
Comment (vkozio) [reply jest.config.js:3]: Because jest replaced by vitest

**Comment by vkozio** [comment]
@ibgreen thanks a lot for feedback. I've opened draft PR to show basically where we are.
Will be great to hear your opinion concerning what should be implemented in scope of this migration task to move us to release.
Thanks.

File: modules/edit-modes/test/lib/__snapshots__/measure-angle-mode.test.ts.snap (hunk 4 above)
Comment (Akiyamka) [reply modules/edit-modes/test/lib/__snapshots__/measure-angle-mode.test.ts.snap:4]: > Some jest version upgrade? 
[vitest](https://vitest.dev/) it's compatible replacement for jest. 
[Here](https://vitest.dev/guide/comparisons.html#jest) is a description of why jest does not work as well in vite as vitest

File: examples/advanced/src/example.tsx [hunk 6]
```diff
@@ -828,7 +830,7 @@ export default class Example extends React.Component<
 
   renderStaticMap(viewport: Record<string, any>) {
     return (
-      // @ts-ignore
+      // @ ts-ignore
```
Comment (Akiyamka) [inline examples/advanced/src/example.tsx:833]: Looks like `@ts-ignore ` it was right
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html

File: examples/advanced/src/example.tsx [hunk 7]
```diff
@@ -1125,7 +1127,7 @@ export default class Example extends React.Component<
                 type: MapController,
                 doubleClickZoom: false,
               },
-              legacyMeterSizes: true,
+              // legacyMeterSizes: true,
```
Comment (Akiyamka) [inline examples/advanced/src/example.tsx:1130]: Why it's commented?

File: examples/advanced/src/example.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
 /* eslint-env browser */
 
-import window from 'global/window';
+// import window from 'global/window';
```
Comment (Akiyamka) [inline examples/advanced/src/example.tsx:3]: I think you can just delete this line

File: examples/advanced/src/index.html [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE html>
+<script type="module">
+  import './app.tsx';
+</script>
```
Comment (Akiyamka) [inline examples/advanced/src/index.html:4]: Pretty unusual, maybe replace it with more common 
`<script type="module" src="./app.tsx"></script>` ?

File: examples/vite.config.local.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
+// This file contains webpack configuration settings that allow
+// examples to be built against the nebula.gl source code in this repo instead
+// of building against their installed version of nebula.gl.
+//
+// This enables using the examples to debug the main nebula.gl library source
+// without publishing or npm linking, with conveniences such hot reloading etc.
+
+// avoid destructuring for older Node version support
+// test
```
Comment (Akiyamka) [inline examples/vite.config.local.ts:9]:
```suggestion
// This file contains configuration settings that allow
// examples to be built against the nebula.gl source code in this repo instead
// of building against their installed version of nebula.gl.
//
// This enables using the examples to debug the main nebula.gl library source
// without publishing or npm linking, with conveniences such hot reloading etc.
```

File: modules/edit-modes/src/index.ts [hunk 11]
```diff
@@ -1,3 +1,8 @@
+import * as utils from './utils';
+
+// Utils
+export { utils };
```
Comment (Akiyamka) [inline modules/edit-modes/src/index.ts:4]: looks redundant

File: modules/edit-modes/src/index.ts [hunk 12]
```diff
@@ -1,3 +1,8 @@
+import * as utils from './utils';
+
+// Utils
```
Comment (Akiyamka) [inline modules/edit-modes/src/index.ts:3]: looks redundant

File: modules/edit-modes/src/index.ts [hunk 13]
```diff
@@ -84,7 +89,3 @@ export type {
   FeatureCollection,
   AnyGeoJson,
 } from './geojson-types';
-
-// Utils
```
Comment (Akiyamka) [inline modules/edit-modes/src/index.ts:88]: looks redundant

File: modules/layers/test/dependencies.test.ts [hunk 14]
```diff
@@ -1,3 +1,5 @@
+//
```
Comment (Akiyamka) [inline modules/layers/test/dependencies.test.ts:1]:
```suggestion
```

File: modules/overlays/test/html-overlay.test.tsx [hunk 15]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (Akiyamka) [inline modules/overlays/test/html-overlay.test.tsx:2]:
```suggestion
```

**Review by Akiyamka — APPROVED** [review]

**Review by Akiyamka — COMMENTED** [review]
`Error: Failed to resolve entry for package "nebula.gl". The package may have incorrect main/module/exports specified in its package.json`

Full log:

```
╭─akiyamka at fedora in ~/git/temp
╰─○ git clone git@github.com:konturio/nebula.gl.git
Cloning into 'nebula.gl'...
remote: Enumerating objects: 7871, done.
remote: Counting objects: 100% (759/759), done.
remote: Compressing objects: 100% (430/430), done.
remote: Total 7871 (delta 426), reused 584 (delta 325), pack-reused 7112
Receiving objects: 100% (7871/7871), 10.25 MiB | 8.47 MiB/s, done.
Resolving deltas: 100% (5127/5127), done.
╭─akiyamka at fedora in ~/git/temp
╰─○ cd nebula.gl          
╭─akiyamka at fedora in ~/git/temp/nebula.gl on master✔
╰─± git checkout 13709-migrate-nebula-to-vite-and-vitest 
branch '13709-migrate-nebula-to-vite-and-vitest' set up to track 'origin/13709-migrate-nebula-to-vite-and-vitest'.
Switched to a new branch '13709-migrate-nebula-to-vite-and-vitest'
╭─akiyamka at fedora in ~/git/temp/nebula.gl on 13709-migrate-nebula-to-vite-and-vitest✔
╰─± code .
╭─akiyamka at fedora in ~/git/temp/nebula.gl on 13709-migrate-nebula-to-vite-and-vitest✔
╰─± yarn
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning " > @deck.gl/extensions@8.8.23" has unmet peer dependency "@luma.gl/constants@^8.0.0".
warning " > @deck.gl/extensions@8.8.23" has unmet peer dependency "@luma.gl/core@^8.0.0".
warning " > @deck.gl/extensions@8.8.23" has unmet peer dependency "gl-matrix@^3.0.0".
warning " > @deck.gl/geo-layers@8.8.23" has unmet peer dependency "@loaders.gl/core@^3.0.0".
warning " > @deck.gl/geo-layers@8.8.23" has unmet peer dependency "@luma.gl/core@^8.0.0".
warning "@deck.gl/geo-layers > @loaders.gl/3d-tiles@3.2.13" has unmet peer dependency "@loaders.gl/core@^3.2.0".
warning "@deck.gl/geo-layers > @loaders.gl/tiles@3.2.13" has unmet peer dependency "@loaders.gl/core@^3.2.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@loaders.gl/gltf@^3.0.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@loaders.gl/images@^3.0.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@luma.gl/engine@^8.4.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@luma.gl/gltools@^8.4.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@luma.gl/shadertools@^8.4.0".
warning "@deck.gl/geo-layers > @luma.gl/experimental@8.5.18" has unmet peer dependency "@luma.gl/webgl@^8.4.0".
warning " > @deck.gl/layers@8.8.23" has unmet peer dependency "@loaders.gl/core@^3.0.0".
warning " > @deck.gl/layers@8.8.23" has unmet peer dependency "@luma.gl/core@^8.0.0".
warning " > @deck.gl/mesh-layers@8.8.23" has unmet peer dependency "@luma.gl/core@^8.0.0".
warning " > babel-loader@9.1.2" has unmet peer dependency "webpack@>=5".
warning " > eslint-config-uber-es2015@3.1.2" has incorrect peer dependency "eslint@>= 3.0.0 < 5".
warning "eslint-config-uber-es2015 > eslint-config-uber-es5@2.0.3" has incorrect peer dependency "eslint@>= 3.0.0 < 5".
warning " > eslint-config-uber-jsx@3.3.3" has incorrect peer dependency "eslint@>= 3.0.0 < 5".
warning "eslint-config-uber-jsx > eslint-plugin-react@6.10.3" has incorrect peer dependency "eslint@^2.0.0 || ^3.0.0".
warning " > ts-loader@9.4.2" has unmet peer dependency "webpack@^5.0.0".
warning " > @nebula.gl/layers@1.0.4" has unmet peer dependency "@luma.gl/constants@8.5.18".
warning " > @nebula.gl/layers@1.0.4" has unmet peer dependency "@luma.gl/core@8.5.18".
warning " > nebula.gl@1.0.4" has unmet peer dependency "@luma.gl/constants@^8.5.18".
warning " > nebula.gl@1.0.4" has unmet peer dependency "@luma.gl/core@^8.5.18".
warning " > react-map-gl-draw@1.0.4" has unmet peer dependency "react-map-gl@^5.x".
warning "workspace-aggregator-56fca823-080a-41c8-8968-58958102948b > @nebula.gl/editor > styled-react-modal@1.2.4" has incorrect peer dependency "react@^16.4.2".
warning "workspace-aggregator-56fca823-080a-41c8-8968-58958102948b > @nebula.gl/editor > styled-react-modal@1.2.4" has incorrect peer dependency "react-dom@^16.4.2".
[4/4] Building fresh packages...
$ yarn run build
yarn run v1.22.19
$ lerna run build --stream
lerna notice cli v6.4.1

 >  Lerna (powered by Nx)   Running target build for 7 projects:

    - example-advanced
    - @nebula.gl/edit-modes
    - @nebula.gl/editor
    - @nebula.gl/layers
    - nebula.gl
    - @nebula.gl/overlays
    - react-map-gl-draw

 —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> @nebula.gl/edit-modes:build


> example-advanced:build


> @nebula.gl/overlays:build

(node:80648) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:80648) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(node:80648) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(node:80648) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
$ vite build
$ vite build -c ../../vite.config.ts
$ vite build -c ../../vite.config.ts
example-advanced: vite v4.0.4 building for production...
example-advanced: transforming...
example-advanced: ✓ 5 modules transformed.
example-advanced: [commonjs--resolver] Failed to resolve entry for package "nebula.gl". The package may have incorrect main/module/exports specified in its package.json.
example-advanced: error during build:
example-advanced: Error: Failed to resolve entry for package "nebula.gl". The package may have incorrect main/module/exports specified in its package.json.
example-advanced:     at packageEntryFailure (file:///home/akiyamka/git/temp/nebula.gl/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:21837:11)
example-advanced:     at resolvePackageEntry (file:///home/akiyamka/git/temp/nebula.gl/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:21834:5)
example-advanced:     at tryNodeResolve (file:///home/akiyamka/git/temp/nebula.gl/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:21572:20)
example-advanced:     at Object.resolveId (file:///home/akiyamka/git/temp/nebula.gl/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:21336:28)
example-advanced:     at file:///home/akiyamka/git/temp/nebula.gl/node_modules/rollup/dist/es/shared/rollup.js:23704:40
example-advanced:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
example-advanced:     at async PluginDriver.hookFirstAndGetPlugin (file:///home/akiyamka/git/temp/nebula.gl/node_modules/rollup/dist/es/shared/rollup.js:23604:28)
example-advanced:     at async resolveId (file:///home/akiyamka/git/temp/nebula.gl/node_modules/rollup/dist/es/shared/rollup.js:22547:26)
example-advanced:     at async ModuleLoader.resolveId (file:///home/akiyamka/git/temp/nebula.gl/node_modules/rollup/dist/es/shared/rollup.js:22811:15)
example-advanced:     at async Object.resolveId (file:///home/akiyamka/git/temp/nebula.gl/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:8001:10)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@nebula.gl/overlays: vite v4.0.4 building for production...
@nebula.gl/edit-modes: vite v4.0.4 building for production...
@nebula.gl/overlays: transforming...
@nebula.gl/edit-modes: transforming...
@nebula.gl/overlays: ✓ 19 modules transformed.
@nebula.gl/overlays: rendering chunks...
@nebula.gl/overlays: computing gzip size...
@nebula.gl/overlays: dist/index.mjs  66.98 kB │ gzip: 18.90 kB │ map: 196.06 kB
@nebula.gl/overlays: dist/index.js  41.29 kB │ gzip: 14.86 kB │ map: 190.81 kB
@nebula.gl/overlays: [vite:dts] Start generate declaration files...
@nebula.gl/edit-modes: ✓ 193 modules transformed.
@nebula.gl/edit-modes: rendering chunks...
@nebula.gl/edit-modes: computing gzip size...
@nebula.gl/overlays: [vite:dts] Start rollup declaration files...
@nebula.gl/edit-modes: dist/index.mjs  652.50 kB │ gzip: 136.21 kB │ map: 2,024.25 kB
@nebula.gl/edit-modes: dist/index.js  489.04 kB │ gzip: 118.57 kB │ map: 1,992.07 kB
@nebula.gl/edit-modes: [vite:dts] Start generate declaration files...
@nebula.gl/edit-modes: [vite:dts] Start rollup declaration files...
@nebula.gl/overlays: [vite:dts] Declaration files built in 8220ms.
@nebula.gl/edit-modes: [vite:dts] Declaration files built in 8330ms.

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  Lerna (powered by Nx)   Running target build for 7 projects failed

   Tasks not run because their dependencies failed or --nx-bail=true:
   
   - @nebula.gl/editor:build
   - @nebula.gl/layers:build
   - nebula.gl:build
   - react-map-gl-draw:build
   
   Failed tasks:
   
   - example-advanced:build

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

```

File: examples/advanced/src/example.tsx (hunk 7 above)
Comment (vkozio) [reply examples/advanced/src/example.tsx:1130]: error TS2345 'legacyMeterSizes' does not exist in type 'MapViewProps & CommonViewProps<MapViewState>'

File: examples/advanced/src/example.tsx (hunk 6 above)
Comment (vkozio) [reply examples/advanced/src/example.tsx:833]: ignore not needed, removed

**Comment by Akiyamka** [comment]
lgfm
